.github/workflows/ci_linux_arm64_clang.yml

```diff
@@ -27,39 +27,44 @@ jobs:
   setup:
     uses: ./.github/workflows/setup.yml
 
-  # TODO(#18238): migrate to new runner cluster
-  #   Or switch to GitHub-hosted arm64 runners when available?
   linux_arm64_clang:
     needs: setup
-    runs-on:
-      - self-hosted # must come first
-      - runner-group=${{ needs.setup.outputs.runner-group }}
-      - environment=${{ needs.setup.outputs.runner-env }}
-      - arm64
-      - os-family=Linux
+    # See https://gitlab.arm.com/tooling/gha-runner-docs
+    runs-on: ah-ubuntu_22_04-c7g_4x-50
+    container:
+      image: ghcr.io/iree-org/cpubuilder_ubuntu_jammy@sha256:f3e2c56680b6007f29c8e191d93dfcb4f8800802e3caeb4851da90f2a1a0b5a5
+    defaults:
+      run:
+        shell: bash
     env:
       BUILD_DIR: build-arm64
     steps:
       - name: "Checking out repository"
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
         with:
           submodules: true
-      - name: "Building IREE"
-        env:
-          IREE_WRITE_REMOTE_CCACHE: ${{ needs.setup.outputs.write-caches }}
+      - name: Install Python requirements
+        run: python3 -m pip install -r ./runtime/bindings/python/iree/runtime/build_requirements.txt
+      # TODO(#18557): enable sccache (watching for network bandwidth charges
+      #               between the cache server and AWS VMs)
+      - name: CMake - configure
         run: |
-          ./build_tools/github_actions/docker_run.sh \
-            --env "IREE_CCACHE_GCP_TOKEN=$(gcloud auth application-default print-access-token)" \
-            --env "IREE_WRITE_REMOTE_CCACHE=${IREE_WRITE_REMOTE_CCACHE}" \
-            --env "CCACHE_NAMESPACE=ci_linux_arm64_clang" \
-            --env "IREE_BUILD_SETUP_PYTHON_VENV=${BUILD_DIR}/.venv" \
-            ghcr.io/iree-org/cpubuilder_ubuntu_jammy@sha256:f3e2c56680b6007f29c8e191d93dfcb4f8800802e3caeb4851da90f2a1a0b5a5 \
-            ./build_tools/cmake/build_all.sh \
-            "${BUILD_DIR}"
-      - name: "Testing IREE"
+          cmake \
+            -G Ninja \
+            -B ${BUILD_DIR} \
+            -DPython3_EXECUTABLE="$(which python3)" \
+            -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+            -DIREE_BUILD_PYTHON_BINDINGS=ON \
+            -DIREE_ENABLE_LLD=ON \
+            -DIREE_ENABLE_ASSERTIONS=ON \
+            -DIREE_BUILD_DOCS=ON \
+            -DIREE_TARGET_BACKEND_WEBGPU_SPIRV=ON
+      - name: CMake - build
         run: |
-          ./build_tools/github_actions/docker_run.sh \
-            --env "IREE_ARM_SME_QEMU_AARCH64_BIN=/usr/bin/qemu-aarch64" \
```

Comment:
Do we need QEMU for SME / SVE at all now? It looks like these runners include those instructions natively?

ctest shows 8 tests tagged `requires-arm-sme` running: https://github.com/iree-org/iree/actions/runs/11114007934/job/30879598592#step:7:3201, but I'm not sure if those are silently passing or what.

Reply:
Okay, for now I'll add back the `IREE_ARM_SME_QEMU_AARCH64_BIN=/usr/bin/qemu-aarch64` environment variable and then check the logs to see if that test takes longer to run.

Longer term, I'm trying to move those matmul tests to the new test suite repository at https://github.com/iree-org/iree-test-suites/tree/main/linalg_ops, where we'll have more direct control over test configurations by being decoupled from the core project build system.

Reply:
I think that worked:

Before:
* https://github.com/iree-org/iree/actions/runs/11114007934/job/30879598592#step:7:2021
* ` 998/1515 Test #1056: iree/tests/e2e/matmul/e2e_matmul_cpu_arm_sme_nondt_f32_large_llvm-cpu_local-task_sme .........................   Passed    0.03 sec`

After:
* https://github.com/iree-org/iree/actions/runs/11128634554/job/30923840594?pr=18643#step:7:2252
* `1113/1515 Test #1056: iree/tests/e2e/matmul/e2e_matmul_cpu_arm_sme_nondt_f32_large_llvm-cpu_local-task_sme .........................   Passed    3.56 sec`